app/controllers/collections_controller.rb

```diff
@@ -170,4 +171,25 @@ def dataset_download
     end
     send_file zip_file
   end
+
+  def build_citation collection
+    return nil unless (!collection.identifier.blank? && collection.identifier.first.include?("doi:"))
+    datacite_api_base = "https://api.datacite.org/application/vnd.schemaorg.ld+json/"
+    datacite_id = collection.identifier.first.gsub("doi:", "")
```

Comment:
You might want to strip the whitespaces at the beginning and the end of the doi string. Because I copied the doi over with an extra ending whitespace which causes the citation to return nil.